cmd/frontend/internal/inventory/files.go

```
@@ -0,0 +1,65 @@
+package inventory
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+// file computes the inventory of a single file. If file is not a regular file, it panics. It caches
```

why panic when you can return an error?

---

Because it means it's being called incorrectly:

```
// Call (*Context).Tree instead, which is much more efficient for computing tree
// inventories because it caches at the tree level.
```

I will add this comment to the docstring instead of putting it in the body.

---

Cleaned up